### PR TITLE
Only upload translations on merge, Create a pull on available updates

### LINKED ISSUE #309: Update Translations via CI
- push en_us.json to crowdin if changed
- pull translations from crowdin & commit if something changed

TODO:
 * [x] create update script (#368)
 * [x] create docker image with `crowdin` cli tool & `nodejs` exec script in CI
 * [x] add secrets & task to CI (#466)

### PR DESCRIPTION
**Pull request by 6543**
fix #309

### CODE REVIEW

File: .github/workflows/pull-translations.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+name: pull-translations
+# pull locale files from crowdin and create a pull if there are changes
+
+on:
+  schedule:
+    # exec each midnight
```
Comment (crapStone) [inline .github/workflows/pull-translations.yml:6]:
```suggestion
    # exec every day at midnight
```

File: .github/workflows/pull-translations.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+      - name: install prettier
+        run: yarn install
+
+      - name: Run Crowdin Update
+        shell: bash
+        env:
+          CROWDIN_TOKEN: ${{ secrets.CROWDIN_TOKEN }}
+        run: ./contrib/pull-translations "$CROWDIN_TOKEN"
+
+      - name: Create Pull Request
```
Comment (crapStone) [inline .github/workflows/pull-translations.yml:26]: Could you close old open PRs or update the old one?

File: contrib/push-translations [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+################################
+# push locale files to crowdin #
+################################
+
+cd "$(dirname "$0")/.."
+token="$1"
+
+# make config if not exist
+[ -f "crowdin.yml" ] || {
+  # make sure it exist
```
Comment (crapStone) [inline contrib/push-translations:12]:
```suggestion
  # make sure it exists
```

**Review by crapStone — CHANGES_REQUESTED** [review]

File: .github/workflows/pull-translations.yml (hunk 2 above)
Comment (6543) [reply .github/workflows/pull-translations.yml:26]: dont know did create an issue upstream ... lets ajust this case later if it doesnt do what we want.

`https://github.com/peter-evans/create-pull-request/issues/893`

Comment (crapStone) [reply .github/workflows/pull-translations.yml:26]: open a followup issue

**Review by crapStone — APPROVED** [review]

**Pull request merged**